src/lib/crypto/kimchi_bindings/stubs/src/caml/caml_pointer.rs

```diff
@@ -58,7 +58,7 @@ macro_rules! impl_caml_pointer {
                     //   mutable, since we can call [`get_mut_unchecked`] in
                     //   nightly, or can call [`get_mut`] and unwrap if this is
                     //   the only live reference.
-                    &mut *(((&*self.0) as *const Self::Target) as *mut Self::Target)
```

Comment:
The comment above probably needs to be updated.

Reply:
Note: as mentioned [here](https://github.com/MinaProtocol/mina/pull/16416#discussion_r1883815674).